AK/StringBuilder.h

```diff
@@ -98,9 +100,10 @@ class StringBuilder {
 
 private:
     ErrorOr<void> will_append(size_t);
-    u8* data() { return m_buffer.data(); }
-    u8 const* data() const { return m_buffer.data(); }
+    u8* data();
+    u8 const* data() const;
 
+    FixedByteBuffer m_fixed_buffer;
     static constexpr size_t inline_capacity = 256;
     Detail::ByteBuffer<inline_capacity> m_buffer;
```

Comment:
StringBuilder already has a fixed size, inline buffer size of 256. Do you really need more bytes than that?

You could avoid needing the FixedStringBuffer class entirely if you annotate the `vformatted` and friends methods on FixedStringBuffer with `requires (Size > StringBuilder::inline_capacity)`